oshdb/src/main/java/org/heigit/ohsome/oshdb/impl/osh/OSHNodeImpl.java

```diff
@@ -73,16 +78,45 @@ public static OSHNodeImpl instance(final byte[] data, final int offset, final in
     // TODO maybe better to store number of versions instead
     final int dataLength = length - (dataOffset - offset);
 
-    return new OSHNodeImpl(data, offset, length, baseNodeId, baseTimestamp, baseLongitude, baseLatitude,
-        header, id, bbox, keys, dataOffset, dataLength);
+    return new OSHNodeImpl(data, offset, length, 
+        baseNodeId, baseTimestamp, baseLongitude, baseLatitude,
+        header, id, minLon, minLat, maxLon, maxLat, keys, dataOffset, dataLength);
   }
 
   private OSHNodeImpl(final byte[] data, final int offset, final int length, final long baseNodeId,
       final long baseTimestamp, final long baseLongitude, final long baseLatitude,
-      final byte header, final long id, final OSHDBBoundingBox bbox, final int[] keys,
+      final byte header, final long id, long minLon, long minLat, long maxLon, long maxLat, 
+      final int[] keys,
       final int dataOffset, final int dataLength) {
     super(data, offset, length, baseNodeId, baseTimestamp, baseLongitude, baseLatitude, header, id,
-        bbox, keys, dataOffset, dataLength);
+        minLon, minLat, maxLon, maxLat, keys, dataOffset, dataLength);
+    
+    // correct bbox!
+    if (!(minLon <= maxLon && minLat <= maxLat)) {
+      minLon = Long.MAX_VALUE;
+      maxLon = Long.MIN_VALUE;
+      minLat = Long.MAX_VALUE;
+      maxLat = Long.MIN_VALUE;
+      for (OSMNode osm : this) {
+        if (osm.isVisible()) {
+          minLon = Math.min(minLon, osm.getLon());
+          maxLon = Math.max(maxLon, osm.getLon());
+
+          minLat = Math.min(minLat, osm.getLat());
+          maxLat = Math.max(maxLat, osm.getLat());
+        }
+      }
+
+      if (minLon == Long.MAX_VALUE || minLat == Long.MAX_VALUE) {
+        minLon = minLat = -1;
+        maxLon = maxLat = 1;
```

Comment:
what does this do?

1. the variables `minLon` etc. are set here, but never used. see [spotbugs warning](https://jenkins.ohsome.org/job/oshdb/view/change-requests/job/PR-369/27/spotbugs/moduleName.736606877/packageName.-1331010043/fileName.1096615460/category.79242641/source.35911c19-2c3c-4bc7-b62c-1240086c5e55/#112)
2. did you mean to set `this.minLon` (etc.) to these values? if yes, I don't necessarily see how setting it to a `(-1,-1,1,1)` bbox around null island makes much sense… can you elaborate please?

Reply:
good catch!